pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -67,8 +79,7 @@ type LocalPortOpener interface {
 type localPortOpener struct{}
 
 type PortTable struct {
-	NodePortTable    map[string]*NodePortData
-	PodEndpointTable map[string]*NodePortData
+	PortTableCache   cache.Store
```

Comment:
Use `cache.Indexer` instead of cache.Store, and define multiple indexers/indexFunc including nodePortProto, and podIPPortProto. Maybe also including podIP as indexer. The key could be nodeportPodIPPortProto to ensure a key is mapping to a unique item NodePortData.

Reply:
Updated.